pkg/spanconfig/target.go

```diff
@@ -53,12 +71,27 @@ func (t Target) String() string {
 // IsEmpty returns true if the receiver is an empty target.
 func (t Target) IsEmpty() bool {
 	return t.GetSpan().Equal(roachpb.Span{})
+
+}
+
+func (t Target) SpanConfigTargetProto() roachpb.SpanConfigTarget {
+	if t.GetSpan() != nil {
+		return roachpb.SpanConfigTarget{
+			Union: &roachpb.SpanConfigTarget_Span{
+				Span: t.GetSpan(),
+			},
+		}
+	}
+
+	panic("cannot handle any other type of target")
 }
 
 // DecodeTarget takes a raw span and decodes it into a Target given its
 // encoding. It is the inverse of Encode.
 func DecodeTarget(sp roachpb.Span) Target {
-	return Target(sp)
+	return Target{
```

Comment:
nit: i'd collapse this to one line, here and throughout but its a purely aesthetic comment so feel free to ignore.